### ISSUE TITLE
Support multiple certificate types simultaneously (RSA/ECDSA/etc.)

### ISSUE THREAD

**Issue by triska** [SWI-Prolog/packages-ssl#81]
There are different **types** of certificates, most notably:

- **RSA** (currently used by the overwhelming majority of web sites)
- **ECDSA** (based on elliptic curves, currently used by a much smaller percentage of web sites)
- **DSA**
- etc.

OpenSSL&ge;**1.0.2** (released in February 2015) supports different certificate chains *per certificate&nbsp;type*, and hence multiple certificate types simultaneously, which may also be issued by different&nbsp;CAs.

I will make this available in SWI-Prolog by a new `ssl_context/3` option like `certificate_key_pairs/1`, where you can specify a **list** of certificates and corresponding&nbsp;keys.

In addition to providing the increasingly important functionality of multiple certificate types, this will also have the benefit of removing the currently obligatory status of the `--keyfile` and `--certfile` arguments for the HTTP Unix&nbsp;daemon. This is useful if you want to rely *entirely* on&nbsp;SNI and drop clients that do not support&nbsp;it.

I will work on this feature in the 7.5 development line, as soon as the decision is made to require at least OpenSSL&nbsp;1.0.2 for building&nbsp;SWI-Prolog.

For making this decision, please take the OpenSSL [**release strategy**](https://www.openssl.org/policies/releasestrat.html) into account:

- Support for version 1.0.1 will **cease on 2016-12-31**. No further releases of 1.0.1 will be made after that date. Security fixes only will be applied to 1.0.1 until then.
- Version 1.0.0 is **no longer supported**.
- Version 0.9.8 is **no longer supported**.

In other words, in a few weeks, said version 1.0.2 will be the **oldest** release of&nbsp;OpenSSL that is still supported.

**Comment by triska** [comment]
I will try to include this in 7.4.

The reason is that we need a more general API in any case to support multiple certificates. The current `certificate/1` and `key/1` options are so new (and only used by the HTTP Unix daemon) that I would prefer one more change to `certificate_key_pairs/1` over introducing a further new option in the next development line.

With older OpenSSL versions, I can make it work in such a way that several types of certificates are admissible as long as the&nbsp;CA is the same for each certificate. Starting with OpenSSL 1.0.2, also completely independent chains will automatically become possible.

**Comment by triska** [comment]
Done!
